EmotiBitOscilloscope/src/ofApp.cpp

```diff
@@ -52,6 +53,121 @@ void ofApp::update() {
 	updateMenuButtons();
 }
 
+void ofApp::setTypeTagPlotAttributes()
+{
+	// ToDo: Add attributes for all streams for refactor
+	{
+		// Add plot attributes for THERMOPILE data
+		typeTagPlotAttr attr;
+		attr.tapeTagName = "THERM";
+		attr.typeTagColor = ofColor(239, 97, 82);
+		// ToDo: someday, we can even consider a standard layout considering even plot number p -> {w,s,p}
+		std::vector<int> sIdx = { 1, 3 };  // {window, scope}
+		attr.scopeIdx = sIdx;
+		typeTagPlotAttributes.emplace(EmotiBitPacket::TypeTag::THERMOPILE, attr);
+	}
+
+	{
+		// Plot Attributes for TEMP1 data
+		typeTagPlotAttr attr;
+		attr.tapeTagName = "TEMP1";
+		attr.typeTagColor = ofColor(234, 174, 68);
+		std::vector<int> sIdx = { 1, 3 };  // {window, scope}
+		attr.scopeIdx = sIdx;
+		typeTagPlotAttributes.emplace(EmotiBitPacket::TypeTag::TEMPERATURE_1, attr);
+	}
+}
+
+void ofApp::resetScopePlot(int w, int s)
+{
+	scopeWins.at(w).scopes.at(s).clearData();
+	scopeWins.at(w).scopes.at(s).setup(10, samplingFreqs.at(w).at(s), plotNames.at(w).at(s), plotColors.at(w).at(s),
+		0, 1);
+}
+
+void ofApp::reinitMetaDataBuffers()
+{
+	bufferSizes = initBuffer(bufferSizes);
+	dataCounts = initBuffer(dataCounts);
+	dataFreqs = initBuffer(dataFreqs);
+}
+
+void ofApp::resetIndexMapping()
+{
+	for (int w = 0; w < typeTags.size(); w++) {
+		for (int s = 0; s < typeTags.at(w).size(); s++) {
+			for (int p = 0; p < typeTags.at(w).at(s).size(); p++) {
+				vector<int> indexes{ w, s, p };
+				typeTagIndexes.emplace(typeTags.at(w).at(s).at(p), indexes);
+			}
+		}
+	}
+}
+
+
+
+void ofApp::addDataStream(std::string typetag)
+{
+	if (typeTagIndexes.find(typetag) == typeTagIndexes.end())
+	{
+		// ToDo: Find a more elegant way to solve default plot state.
+		// If adding THERM, remove default TEMP1. It will be added by separate add() call if Temp1 stream is found
+		if (typetag == EmotiBitPacket::TypeTag::THERMOPILE)
+		{
+			removeDataStream(EmotiBitPacket::TypeTag::TEMPERATURE_1);
```

Comment:
`TEMP1` is added to the scope as a default place holder because some parts of the code reference `typeTags`(`vector<vector<vector>>>`) to update the plots. In the future, I want to work on creating a default oscilloscope (on startup), which gets updated with plots after data is detected and moves away from the tripple-vector.
When I add `THERM`, I remove the default since we are not sure if that data is present. It gets added back to the plot when data of type `T1` is processed. I added a more detailed comment explaining the reason.